Traceback (most recent call last):
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/convert_frame.py", line 133, in _fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/convert_frame.py", line 389, in _convert_frame_assert
    return _compile(
           ^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/convert_frame.py", line 569, in _compile
    guarded_code = compile_inner(code, one_graph, hooks, transform)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/utils.py", line 189, in time_wrapper
    r = func(*args, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/convert_frame.py", line 491, in compile_inner
    out_code = transform_code_object(code, transform)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/bytecode_transformation.py", line 1028, in transform_code_object
    transformations(instructions, code_options)
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/convert_frame.py", line 458, in transform
    tracer.run()
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/symbolic_convert.py", line 2074, in run
    super().run()
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/symbolic_convert.py", line 724, in run
    and self.step()
        ^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/symbolic_convert.py", line 688, in step
    getattr(self, inst.opname)(inst)
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/symbolic_convert.py", line 2162, in RETURN_VALUE
    self.output.compile_subgraph(
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/output_graph.py", line 857, in compile_subgraph
    self.compile_and_call_fx_graph(tx, pass2.graph_output_vars(), root)
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/contextlib.py", line 81, in inner
    return func(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/output_graph.py", line 957, in compile_and_call_fx_graph
    compiled_fn = self.call_user_compiler(gm)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/utils.py", line 189, in time_wrapper
    r = func(*args, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/output_graph.py", line 1024, in call_user_compiler
    raise BackendCompilerFailed(self.compiler_fn, e).with_traceback(
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/output_graph.py", line 1009, in call_user_compiler
    compiled_fn = compiler_fn(gm, self.example_inputs())
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/backends/distributed.py", line 436, in compile_fn
    submod_compiler.run(*example_inputs)
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/fx/interpreter.py", line 138, in run
    self.env[node] = self.run_node(node)
                     ^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/backends/distributed.py", line 417, in run_node
    compiled_submod_real = self.compile_submod(
                           ^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/backends/distributed.py", line 361, in compile_submod
    self.compiler(input_mod, args),
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/repro/after_dynamo.py", line 117, in debug_wrapper
    compiled_gm = compiler_fn(gm, example_inputs)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/__init__.py", line 1568, in __call__
    return compile_fx(model_, inputs_, config_patches=self.config)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_inductor/compile_fx.py", line 1150, in compile_fx
    return aot_autograd(
           ^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/backends/common.py", line 55, in compiler_fn
    cg = aot_module_simplified(gm, example_inputs, **kwargs)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_functorch/aot_autograd.py", line 3891, in aot_module_simplified
    compiled_fn = create_aot_dispatcher_function(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/utils.py", line 189, in time_wrapper
    r = func(*args, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_functorch/aot_autograd.py", line 3337, in create_aot_dispatcher_function
    with torch.autograd.set_multithreading_enabled(
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_decomp/decompositions_for_rng.py", line 127, in __exit__
    PhiloxStateTracker.reset()
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_decomp/decompositions_for_rng.py", line 131, in reset
    cls.running_state = PhiloxState()
                        ^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_decomp/decompositions_for_rng.py", line 72, in __init__
    self.reset()
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_decomp/decompositions_for_rng.py", line 75, in reset
    self.seed = torch.tensor(())
                ^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/utils/_stats.py", line 20, in wrapper
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_subclasses/fake_tensor.py", line 1248, in __torch_dispatch__
    assert self not in _get_current_dispatch_mode_stack(), func
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
torch._dynamo.exc.BackendCompilerFailed: backend='compile_fn' raised:
AssertionError: aten.lift_fresh.default
While executing %submod_1 : [num_users=2] = call_module[target=submod_1](args = (%getitem, %getitem_1, %l__self___transformer_h_0_ln_2_weight, %l__self___transformer_h_1_ln_1_weight, %l__self___transformer_h_1_ln_2_weight, %l__self___transformer_h_2_ln_1_weight, %l__self___transformer_h_2_ln_2_weight, %l__self___transformer_h_3_ln_1_weight), kwargs = {})
Original traceback:
None
Set TORCH_LOGS="+dynamo" and TORCHDYNAMO_VERBOSE=1 for more information
You can suppress this exception and fall back to eager by setting:
    import torch._dynamo
    torch._dynamo.config.suppress_errors = True
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/home/loki/projects/adgpt/train.py", line 264, in <module>
    losses = estimate_loss()
             ^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/projects/adgpt/train.py", line 224, in estimate_loss
    logits, loss = model(X, Y)
                   ^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/nn/parallel/distributed.py", line 1519, in forward
    else self._run_ddp_forward(*inputs, **kwargs)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/nn/parallel/distributed.py", line 1355, in _run_ddp_forward
    return self.module(*inputs, **kwargs)  # type: ignore[index]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/eval_frame.py", line 328, in _fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/eval_frame.py", line 487, in catch_errors
    return hijacked_callback(frame, cache_entry, hooks, frame_state)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/convert_frame.py", line 641, in _convert_frame
    result = inner_convert(frame, cache_size, hooks, frame_state)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_dynamo/convert_frame.py", line 140, in _fn
    torch.cuda.set_rng_state(cuda_rng_state)
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/cuda/random.py", line 62, in set_rng_state
    new_state_copy = new_state.clone(memory_format=torch.contiguous_format)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/utils/_stats.py", line 20, in wrapper
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_subclasses/fake_tensor.py", line 1250, in __torch_dispatch__
    return self.dispatch(func, types, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_subclasses/fake_tensor.py", line 1376, in dispatch
    ) = self.validate_and_convert_non_fake_tensors(func, converter, args, kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_subclasses/fake_tensor.py", line 1597, in validate_and_convert_non_fake_tensors
    args, kwargs = tree_map_only(
                   ^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/utils/_pytree.py", line 353, in tree_map_only
    return tree_map(map_only(ty)(fn), pytree)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/utils/_pytree.py", line 283, in tree_map
    return tree_unflatten([fn(i) for i in flat_args], spec)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/utils/_pytree.py", line 283, in <listcomp>
    return tree_unflatten([fn(i) for i in flat_args], spec)
                           ^^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/utils/_pytree.py", line 334, in inner
    return f(x)
           ^^^^
  File "/home/loki/anaconda3/envs/rb/lib/python3.11/site-packages/torch/_subclasses/fake_tensor.py", line 1587, in validate
    raise Exception(
Exception: Please convert all Tensors to FakeTensors first or instantiate FakeTensorMode with 'allow_non_fake_inputs'. Found in aten.clone.default(tensor([...], size=(16,), dtype=torch.uint8), memory_format=torch.contiguous_format)